### PR TITLE
Adds an air analyzer to drones

### PR DESCRIPTION
**Pull request by warior4356**
## What Does This PR Do
This PR adds an air analyzer to maintenance drones.

## Why It's Good For The Game
The air analyzer is good because it lets them actually check the air levels in a room or a pipe, which is good when trying to restore hull breaches or locate a broken pipe.

## Changelog
:cl:
add: Added an air analyzer to maintenance drones
/:cl:

### CODE REVIEW

**Comment by SteelSlayer** [comment]
:-1: is due to the blueprints. Air analyzer is fine.

**Comment by Flotilla-zz** [comment]
Giving drones blueprints gives implicit encouragement for them to make significant expansions to the station, which can easily fall afoul of:

>Drones may not attempt to alter the course of the round with their actions.

**Comment by warior4356** [comment]
If a head admin or maintainer decides that the ability for drones to make extensions to the station is the veto point on this PR please let me know. I will happily make a version of blueprints that can't add new areas and can only view what the station is supposed to look like for repairs.

**Comment by S34NW** [comment]
> If a head admin or maintainer decides that the ability for drones to make extensions to the station is the veto point on this PR please let me know. I will happily make a version of blueprints that can't add new areas and can only view what the station is supposed to look like for repairs, as trying to figure out where the piping and wiring was is a pain and maintenance drones probably should try to put a room back the way it was, as close as possible, instead of haphazard fixes.

I am neither of these things, but I really encourage you to implement this "read only" blueprint idea. The last thing anyone wants is the extra workload of dealing with drones who think renaming all the station areas to "maints" is a fun joke.

**Review by dearmochi — APPROVED** [review]
Approved for code. Not a fan of the blueprints, but while you're at it I think drones could use a floor painter.

**Comment by LiberatedWaveMan** [comment]
As everyone else has said by now, the blueprints are _eh_ but the air analyzer is fine. Floor painter would be good as mochi said, and perhaps plasteel for repairing reinforced walls?

**Comment by warior4356** [comment]
I personally think drones having the ability to make rooms would be a positive for the round, not a negative but I'm not in the majority. Still, I'm going to refrain from tweaking the PR unless a head or maint would like it changed.

**Comment by Kyep** [comment]
Do their blueprints let them build new rooms? If so, maybe give them a type that does not allow this?
They can maintain the existing station, but cannot make entirely new areas.

**Comment by warior4356** [comment]
> Do their blueprints let them build new rooms? If so, maybe give them a type that does not allow this?
> They can maintain the existing station, but cannot make entirely new areas.

Alright, I'll up the PR like that.

**Comment by warior4356** [comment]
Alright, a read only version of the blueprints has been given to drones, and set as a new variable within blueprints/area editors in general so it can be scaled as needed for other things, to avoid this being snowflake code. I hope this mollifies people's concerns.

**Comment by Fox-McCloud** [comment]
Blueprints, I'm not a fan of.

Air analyzer is fine.

**Comment by warior4356** [comment]
> Blueprints, I'm not a fan of.
> 
> Air analyzer is fine.

As above, I updated the blueprints into being read-only. They can only view what the station is supposed to look like and is unable to modify rooms or make new ones.

**Review by Kyep — APPROVED** [review]

**Review by variableundefined — CHANGES_REQUESTED** [review]
What Fox said

File: code/game/objects/items/blueprints.dm [hunk 1]
```diff
@@ -39,7 +40,7 @@
 	if(..())
 		return
 	if(href_list["create_area"])
-		if(get_area_type()==AREA_SPACE)
+		if(get_area_type()==AREA_SPACE && !read_only) // Never trust user input
```
Comment (AffectedArc07) [inline code/game/objects/items/blueprints.dm:43]:
```suggestion
		if(get_area_type() == AREA_SPACE && !read_only) // Never trust user input
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Im with the others. 

Remove the blueprints.

**Comment by warior4356** [comment]
Blueprints have been removed.

**Review by variableundefined — APPROVED** [review]

**Pull request merged**